specification/communication/data-plane/PhoneNumbers/preview/2021-10-25-preview/phonenumbers.json

```diff
@@ -0,0 +1,1639 @@
+            "type": "integer",
+            "format": "int32",
+            "default": 0
+          },
+          {
+            "in": "query",
+            "name": "top",
+            "description": "An optional parameter for how many entries to return, for pagination purposes. The default value is 100.",
+            "type": "integer",
+            "format": "int32",
+            "default": 100
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/PurchasedPhoneNumbers"
+            }
+          },
+          "default": {
+            "description": "Failure",
+            "schema": {
+              "$ref": "../../../Common/stable/2021-03-07/common.json#/definitions/CommunicationErrorResponse"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink",
+          "itemName": "phoneNumbers"
+        },
+        "x-ms-examples": {
+          "List purchased phone numbers": {
+            "$ref": "./examples/GetPhoneNumbers.json"
+          }
+        }
+      }
+    },
+    "/shortCodes": {
+      "get": {
+        "tags": [
+          "ShortCodes"
+        ],
+        "summary": "Gets the list of short codes for the current resource.",
+        "operationId": "ShortCodes_GetShortCodes",
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "in": "query",
+            "name": "skip",
+            "description": "An optional parameter for how many entries to skip, for pagination purposes.",
+            "type": "integer",
+            "format": "int32",
+            "default": 0
+          },
+          {
+            "in": "query",
+            "name": "take",
+            "description": "An optional parameter for how many entries to return, for pagination purposes.",
+            "type": "integer",
+            "format": "int32",
+            "default": 100
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/ShortCodeEntities"
+            }
+          },
+          "default": {
+            "description": "Failure",
+            "schema": {
+              "$ref": "../../../Common/stable/2021-03-07/common.json#/definitions/CommunicationErrorResponse"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink",
+          "itemName": "shortCodes"
+        },
+        "x-ms-examples": {
+          "List short codes": {
+            "$ref": "./examples/GetShortCodesAsync.json"
+          }
+        }
+      }
+    },
+    "/shortCodes/countries/US/programBriefs/{programBriefId}": {
```

Comment:
This is kind of interesting...  When we add other values in the future, we'll create other paths with this same shape but different models?  That feels a little funky if we're changing all the model types for different entries in a collection unless we design them to be polymorphic when we add all the rest.

Reply:
You're not wrong that polymorphic models are a bad time.  I think the way you've got the path structured today is potentially problematic for a different reason though.  I worry anyone looking at this path will see `US` as a member of the `countries` collection rather than `countries/US/` being a constant path.  Most Azure REST APIs don't return different, unrelated models for different members of a collection.  If we made an analogy to C#, since `countries` is a collection, I'd expect the type to be `IList<ProgramBrief>` where `UsProgramBrief : ProgramBrief`.  Instead you have `List<object>` where the values will be things like `UsProgramBrief : object`, `NextProgramBrief : object`, etc.  It's not impossible to use, but it departs from most Azure services.

I see two ways to address this:
- If we want some of the other operations like Get/Delete to work the same across all countries, then a polymorphic model is probably the right choice.
- If we think a polymorphic model isn't possible, then let's consider a separate `UsProgramBriefs` collection in the path that's tied to that particular shape.  If you're really intending these to be static, non-parameterized paths then this might be more accurate.

Happy to chat about this with you and Jeff over a quick Teams call too.

Reply:
The reason why I modeled the path this way was to be consistent with the existing endpoint (Already GAd) to search phone numbers which looks like /availablePhoneNumbers/countries/{countryCode}/:search. In that endpoint countries/{countryCode} is not meant to be a collection of countries, it is meant to be just a path param that is used as part of the input that the search operation needs to do its job. Honestly I would have modeled that more like /availablePhoneNumbers/:search?countryCode={countryCode}.

I did consider initially your second alternative of using UsProgramBriefs and I think that is a better modeling from a REST point of view. I just thought that modeling the path the same way than the search endpoint would allow me to not use the country code as a param but just as a constant so that I can have different routes for each country so that each would have their own schema describing its input and output while it would still look the similar.

I'd like to have this chat on Teams, would you have time today? @tg-msft @JeffreyRichter ?

Reply:
I appreciate trying to be consistent but is the search API monomorphic while your API is polymorphic? If so, then this justifies being inconsistent.

Reply:
Hey Jeffrey, my API is not polymorphic, at least not for now. Could we talk on the phone about this? I'd like to drive this discussion to closure and make any necessary changes soon.

Reply:
Hey @tg-msft, @JeffreyRichter and I talked on the phone, we think this is ok this way. Please note that, as I mentioned before, the 'US' part of the path is fixed (not a path param) so when we expand to Canada we will have paths for Canada like /shortCodes/countries/CA/programBriefs for all the different HTTP methods. This implies that the methods on the SDK client will have to have the US or CA on the method names like client.createUSProgramBrief and client.createCAProgramBrief.